pkgs/nix-tools/darwin-rebuild.sh

```diff
@@ -137,6 +141,19 @@ if [ -z "$action" ]; then showSyntax; fi
 
 flakeFlags=(--extra-experimental-features 'nix-command flakes')
 
+# Search the default locations for the system-wide nix-darwin configuration
+if [[ -z $flake && -z $noFlake ]]; then
+  if [[ -e /etc/nix-darwin/flake.nix ]]; then
+    flake="$(dirname "$(readlink -f /etc/nix-darwin/flake.nix)")"
+  elif [[ -e ~/.config/nix-darwin/flake.nix ]]; then
+    flake="$(dirname "$(readlink -f ~/.config/nix-darwin/flake.nix)")"
+  elif [[ -e ~/.nixpkgs/flake.nix ]]; then
+    flake="$(dirname "$(readlink -f ~/.nixpkgs/flake.nix)")"
+  fi
+fi
```

Comment:
None of these were defaulted to previously, right? I’m a bit wary about giving preferential behaviour to directories we want to get rid of – `~/.nixpkgs` especially, that’s a totally inappropriate place for us to be using, especially for a flake setup – and making it more painful for us to do a deprecation process later. I think we could just use `/etc/nix-darwin` here and worry about documentation etc. later.

`~/.config/nix-darwin` makes some amount of sense while the scourge of user activation is still with us, but since nobody’s existing setup is using this feature, I feel like it’s better to take an incremental step forward than one that sanctions directories we’d like to move away from.